README.md

```diff
@@ -137,15 +140,27 @@ In particular, when creating a document or worker, we use the following in place
 
 This algorithm has two interesting features:
 
-* Even if origin isolation is not requested for a particular document/worker creation, if origin isolation has previously created an origin-keyed agent cluster, then we put the new document/worker in that origin-keyed agent cluster.
-* Even when origin isolation is requested, if there is a site-keyed agent cluster with same-origin realms, then we put the new document/worker in that site-keyed agent cluster, ignoring the isolation request.
+* Even if origin isolation is not requested for a particular document creation, if origin isolation has previously created an origin-keyed agent cluster, then we put the new document in that origin-keyed agent cluster.
+* Even when origin isolation is requested, if there is a site-keyed agent cluster with same-origin documents, then we put the new document in that site-keyed agent cluster, ignoring the isolation request.
 
 Both of these are consequences of a desire to ensure that same-origin sites do not end up isolated from each other.
 
 You can see a more full analysis of what results this algorithm produces in our [scenarios document](./scenarios.md).
 
 As far as the web-developer–observable consequences of using origins for agent cluster keys, this will automatically cause `SharedArrayBuffer`s to no longer be shareable cross-origin, because of the agent cluster check in [StructuredDeserialize](https://html.spec.whatwg.org/multipage/structured-data.html#structureddeserialize). We'd also need to add a check to the [`document.domain` setter](https://html.spec.whatwg.org/multipage/origin.html#dom-document-domain) to make it throw. (Ideally we'd find some way of restructuring the existing checks in `document.domain` so that they are also tied to agent cluster boundaries.)
 
+### More detail on workers
+
+We mentioned above that origin isolation has no effect on workers. Here we explain why exactly that is.
+
+First note that shared and service workers are already isolated into their own agent clusters. The only other things that exist in those agent clusters are potential nested dedicated workers. Everything is already same-origin within those shared and service worker agent clusters.
```

Comment:
Not quite everything, but everything that creates an agent, yes?

Reply:
I guess I meant "all other agents are already same-origin", so this should probably be merged with the previous sentence. Will update.